pxtblocks/blocklyloader.ts

```diff
@@ -540,48 +512,117 @@ namespace pxt.blocks {
+                    }
+                  ],
+                  "previousStatement": null,
+                  "nextStatement": null,
+                  "colour": blockColors['loops']
+                });
                 this.appendStatementInput("DO")
-                    .appendField("do");
-                this.setPreviousStatement(true);
-                this.setNextStatement(true);
-                this.setTooltip(lf("Run the same sequence of actions while the condition is met."));
+                    .appendField(lf("_while_do")); // "do" in English
+
+                setHelpResources(this,
+                    'device_while',
+                    lf("a loop that repeats while the condition is true"),
+                    lf("Run the same sequence of actions while the condition is met."),
+                    '/blocks/loops/while'
+                );
             }
         };
 
+        // pxt controls_simple_for
         Blockly.Blocks['controls_simple_for'] = {
             /**
              * Block for 'for' loop.
              * @this Blockly.Block
              */
             init: function () {
-                this.setHelpUrl("/blocks/loops/for");
-                this.setColour((<any>Blockly.Blocks).loops.HUE);
-                this.appendDummyInput()
-                    .appendField("for")
-                    .appendField(new Blockly.FieldVariable(null), 'VAR')
-                    .appendField("from 0 to");
-                this.appendValueInput("TO")
-                    .setCheck("Number")
-                    .setAlign(Blockly.ALIGN_RIGHT);
+                this.jsonInit({
+                  "message0": lf("for %1 from 0 to %2"),
+                  "args0": [
+                    {
+                      "type": "field_variable",
+                      "name": "VAR",
+                      "variable": lf("_default_var") // "item" in English
+                      // Please note that most multilingual characters
+                      // cannot be used as variable name at this point.
+                      // Translate or decide the default variable name
+                      // with care.
+                    },
+                    {
+                      "type": "input_value",
+                      "name": "TO",
+                      "check": "Number"
+                    }
+                  ],
+                  "previousStatement": null,
+                  "nextStatement": null,
+                  "colour": blockColors['loops'],
+                  "inputsInline": true
+                });
                 this.appendStatementInput('DO')
-                    .appendField(Blockly.Msg.CONTROLS_FOR_INPUT_DO);
-                this.setPreviousStatement(true);
-                this.setNextStatement(true);
-                this.setInputsInline(true);
-                // Assign 'this' to a variable for use in the tooltip closure below.
+                    .appendField(lf("_for_do")); // "do" in English
```

Comment:
Same as ``_for_do`` and ``_while_do``, please open an issue and we'll fix this in a future commit.